### PR TITLE
diary: init at 0.3

### PR DESCRIPTION
**Pull request by matthiasbeyer**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/diary/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  src = fetchFromGitHub {
+    owner = "in0rdr";
+    repo = "diary";
+    rev = "v${version}";
+    sha256 = "0sda8hrnhs7zvl10iri6srllcimi719yvzn22g9jmiryi9sllay6";
+  };
+
+  buildInputs = [ ncurses readline ];
+
+  installPhase = ''
```
Comment (pSub) [inline pkgs/applications/misc/diary/default.nix]: Please move the creation of the directories into `preInstalledPhase` and pass `PREFIX=$out` via `installFlags`.

File: pkgs/applications/misc/diary/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    mkdir $out/share/man/ -p
+    PREFIX=$out make install
+  '';
+
+  meta = {
+    homepage = "https://github.com/in0rdr/diary";
+    description = "Simple CLI diary";
+    license = stdenv.lib.licenses.mit;
+    maintainers = [ stdenv.lib.maintainers.matthiasbeyer ];
+    platforms = with stdenv.lib.platforms; linux;
```
Comment (pSub) [inline pkgs/applications/misc/diary/default.nix]: You don't need `with` in that case, it will just increase runtime.

**Review by pSub — CHANGES_REQUESTED** [review]

**Comment by matthiasbeyer** [comment]
@pSub have a look at the last commit, that does not work... I tried `preInstallPhase` and `preInstalledPhase` ...

**Comment by puffnfresh** [comment]
@matthiasbeyer you want `preInstall` instead of `preInstalledPhase`

**Pull request closed without merging**